README.md

```diff
@@ -247,35 +295,80 @@ std::optional<lex_item> lexer(std::string_view v) noexcept {
 This support is preliminary, probably the API will be changed.
 
 ```c++
-auto input = "123,456,768"sv;
+#include <ctre.hpp>
+#include <iostream>
+#include <string_veiw>
 
-for (auto match: ctre::range<"([0-9]+),?">(input)) {
+// auto input = "123,456,768";
+extern const char *input;
+
+int main(void) {
+  auto matches = ctre::range<"([0-9]+),?">(input);
+  for (auto match : matches) {
     std::cout << std::string_view{match.get<0>()} << "\n";
+  }
+  return 0;
 }
 ```
+[link to compiler explorer](https://gcc.godbolt.org/z/s4zedb68n)
 
 ### Unicode
 
 ```c++
 #include <ctre-unicode.hpp>
 #include <iostream>
+
 // needed if you want to output to the terminal
 std::string_view cast_from_unicode(std::u8string_view input) noexcept {
-    return std::string_view(reinterpret_cast<const char *>(input.data()), input.size());
+  return std::string_view(reinterpret_cast<const char *>(input.data()), input.size());
 }
-int main()
-{
-    using namespace std::literals;
-    std::u8string_view original = u8"Tu es un génie"sv;
-
-    for (auto match : ctre::range<"\\p{Letter}+">(original))
-        std::cout << cast_from_unicode(match) << std::endl;
-    return 0;
+
+int main() {
+  using namespace std::literals;
+  std::u8string_view original = u8"Tu es un génie"sv;
+
+  for (auto match : ctre::range<"\\p{Letter}+">(original))
+    std::cout << cast_from_unicode(match) << std::endl;
+  return 0;
 }
 ```
 
 [link to compiler explorer](https://godbolt.org/z/erTshe6sz)
 
+## Integration
+You can get [ctre.hpp](https://github.com/hanickadot/compile-time-regular-expressions/blob/main/single-header/ctre.hpp) from the directory `single-header`.
+You need to add:
+
+```C++
+#include "ctre.hpp"
+
+//using namespace ctre;
+```
+This header can be regenerated with `make single-header`.
+### CMake
+If you are using cmake, you can add this directory as subdirectory and link to target `ctre`.
+
+```CMake
+cmake_minimum_required(VERSION 3.8.0)
+
+project(MyProject VERSION 1.0)
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+FetchContent_Declare(
+        ctre
+        GIT_REPOSITORY https://github.com/hanickadot/compile-time-regular-expressions.git
+        GIT_TAG 95c63867bf0f6497825ef6cf44a7d0791bd25883 # v3.4.1
+)
+
+FetchContent_MakeAvailable(ctre)
```

Comment:
Please read the [docs](https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_makeavailable):

> New in version 3.14

If https://github.com/hanickadot/compile-time-regular-expressions/pull/230 is merged then this becomes entirely wrong, as using ctre would simply be:

```cmake
find_package(ctre REQUIRED)
target_link_libraries(project_target PRIVATE ctre::ctre)
```

Reply:
I think that if the package was not found we could do

(pseudocode)
```
find_package(ctre)
if(package not found){
    FetchContent(ctre)
    include_target_directories(ctre dir)
}
```

Reply:
That's not the responsibility of this project. There are many ways to "polyfill" a `find_package` call. It should instead just use idiomatic CMake in examples.

Reply:
I guess
```suggestion
find_package(ctre REQUIRED) 
target_link_libraries(project_target PRIVATE ctre::ctre)
```